src/rtc/mosaic_geobursts.py

```diff
@@ -0,0 +1,291 @@
+
+import numpy as np
+from osgeo import gdal
+
+
+def check_mosaic_eligibility(list_rtc: list, list_nlooks: list) -> bool:
+    '''
+    Check if the list of the geobursts are eligible to be mosaiced
+
+    Parameters:
+    -----------
+        list_rtc: list
+            path to the RTC geoburst
+        list_nlooks: list
+            path to the nlooks raster that corresponds to list_rtc
+
+    Returns:
+        flag_rtn: bool
+            Flag if the lists of the input rtc and nlooks are eligible for mosaicking.
+    '''
+
+    # Accepted error in the coordinates as floating number. Used when checking the snapping.
+    maxerr_coord = 1.0e-6
+
+    # Return value after this mosaicking eligibility check.
+    # It will turn to False if any of the checks below fails.
+    flag_rtn = True
+
+    # Check the number of the raster files in list_rtc and list_nlooks
+    num_rtc = len(list_rtc)
+    num_nlooks = len(list_nlooks)
+    if num_rtc != num_nlooks:
+        #raise ValueError(f'# RTC ({num_rtc}) and # nlooks ({num_nlooks}) does not match.')
+        print(f'# RTC ({num_rtc}) and # nlooks ({num_nlooks}) does not match.')
+        flag_rtn=False
+
+    else:
+        # Variables to keep record of the geogrid-related information in the input rasters
+        str_proj_prev = None
+        spacing_x_prev = None
+        spacing_y_prev = None
+        numbands_prev = None
+        mod_x_prev = None
+        mod_y_prev = None
+
+        for i, path_rtc in enumerate(list_rtc):
+            path_nlooks = list_nlooks[i]
+
+            raster_rtc = gdal.Open(path_rtc, gdal.GA_ReadOnly)
+            raster_nlooks = gdal.Open(path_rtc, gdal.GA_ReadOnly)
+
+            geo_transformation_rtc = raster_rtc.GetGeoTransform()
+            geo_transformation_nlooks = raster_nlooks.GetGeoTransform()
+
+            # Compare Geotransform - between RTC and corresponding nlooks
+            if geo_transformation_rtc != geo_transformation_nlooks:
+                print('GeoTransform does not match between '+\
+                     f'{os.path.basename(path_rtc)} and {os.path.basename(path_nlooks)}')
+                flag_rtn = False
+
+            # Compare dimension - between RTC and corresponding nlooks
+            if (raster_rtc.RasterXSize != raster_nlooks.RasterXSize) or\
+            (raster_rtc.RasterYSize != raster_nlooks.RasterYSize):
+                print('Dimension does not agree between '+\
+                     f'{os.path.basename(path_rtc)} and {os.path.basename(path_nlooks)}')
+                flag_rtn = False
+
+            # Check number of bands - for all RTC
+            if numbands_prev is None:
+                numbands_prev = raster_rtc.RasterCount
+            else:
+                if numbands_prev != raster_rtc.RasterCount:
+                    print(f'Band number anomaly detected from {os.path.basename(path_rtc)}')
+                    flag_rtn = False
+                #else:
+                #    numbands_prev = raster_rtc.RasterCount
```

Comment:
```suggestion
            elif numbands_prev != raster_rtc.RasterCount:
                print(f'Band number anomaly detected from {os.path.basename(path_rtc)}')
                flag_rtn = False
```

Reply:
Revised on the other if statements. Thanks for the suggestion!